### DISCUSSION TITLE
How to export jsx styled in a github package

### DISCUSSION THREAD

**Discussion by ImJustLucas** [chakra-ui/panda#2601] [Q&A] [answered]
Hello! I have a problem exporting a jsx component from the styled function, which is in a github package

the purpose of this package is to define my panda preset for a total of 3 apps, and also to serve as a mini ui kit, to export my jsx components which derive from recipes. I've already tried putting my component and recipe at the root of a test project(app/playground in the repo), and everything works correctly.

Now the problem is to put it in a github package, I have a problem compiling / pnpm i, how can I do?

[EDIT]: here's the repo: https://github.com/qotum/qotum

i've followed some tutorial, but i dit not success to solve my problem
[1st tutorial](https://panda-css.com/docs/guides/component-library#use-panda-as-external-package) / [2nd tutorial](https://github.com/astahmer/panda-monorepo-setup/blob/main/examples/with-import-map-package/README.md)

I want my external next app to consume `panda-preset`, where `panda-preset` contains shared components, Panda tokens, and semantics token.

and that the error i get when i `pnpm i` my monorepo or my test app with the package installed, or when i `panda codegen` this package

```bash
panda codegen
│ ✘ [ERROR] Could not resolve "../../../styled-system/jsx"
│     src/components/button/index.ts:1:23:
│       1 │ import { styled } from "../../../styled-system/jsx";
│         ╵                        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
│ ✘ [ERROR] Could not resolve "../../../styled-system/recipes"
│     src/components/button/index.ts:2:23:
│       2 │ import { button } from "../../../styled-system/recipes";
```

what can i do to resolve this ?

**Comment by astahmer** [comment, marked as answer]
the main issue was that you have a barrel file (index.ts that re-exports stuff) in your preset that includes both your components AND the preset
so when the panda.config was resolved and saw the preset import, it also was trying to bundle the whole components directory (including the styled-system)

made a PR that fixes this and other stuff here https://github.com/qotum/qotum/pull/1

**Comment by ImJustLucas** [comment]
Thank you so much @astahmer ! 
I have a second question now, let's imagine I want to export the `@panda-preset` package, which is published on the npm github registry (the monorepo is mainly used to export packages) to app next on another repo, given that in the panda.config of app/playground, the outDir value is a path to the package in the monorepo,
`outdir: "../../packages/panda-exports/dist",`
 should I change it to the @qotum/panda-exports package like this?

`outdir: "@qotum/panda-exports/panda-exports/dist",`

**Reply by ImJustLucas** [reply]
nevermind, i just connect my 2 cell brains lol, thanks for the help, t'es le boss
